unit_test/solver/matrix/test_matrix.py

```diff
@@ -53,8 +51,8 @@ def test_add_unknowns(self) -> None:
 
         # assert
         self.assertEqual(matrix.num_unknowns, number_of_unknowns)
-        self.assertEqual(matrix.sol_new, [1.0] * number_of_unknowns)
-        self.assertEqual(matrix.sol_old, [0.0] * number_of_unknowns)
+        self.assertEqual(matrix.sol_new.tolist(), [1.0] * number_of_unknowns)
```

Comment:
You can also test numpy arrays
```suggestion
        import numpy.testing as npt
        npt.assert_equal()
```
because you want to know that it is an array of size (1, N) or (N,1)

Reply:
done